hazelcast/sql.py

```diff
@@ -0,0 +1,1604 @@
+                return
+
+            # The result contains the next page, as expected.
+            self._on_fetch_response(response["row_page"])
+        except Exception as e:
+            # Something went bad, we couldn't get response from
+            # the server, invocation failed.
+            self._on_fetch_error(self._sql_service.re_raise(e, self._connection))
+
+    def _on_fetch_error(self, error):
+        """Sets the fetch future with exception, but not resetting it
+        so that the next fetch request fails immediately.
+
+        Args:
+            error (Exception): The error.
+        """
+        with self._lock:
+            self._fetch_future.set_exception(error)
+
+    def _on_fetch_response(self, page):
+        """Sets the fetch future with the next page,
+        resets it, and if this is the last page,
+        marks the result as closed.
+
+        Args:
+            page (_SqlPage): The next page.
+        """
+        with self._lock:
+            self._fetch_future.set_result(page)
+            if page.is_last:
+                # This is the last page, there is nothing
+                # more on the server.
+                self._mark_closed()
+
+            self._fetch_future = None
+
+    def _handle_execute_response(self, future):
+        """Handles the result of the execute request, by either:
+
+        - setting it to an exception so that the dependent methods
+          (iterator, update_count etc.) fails immediately
+        - setting it to an execute response
+
+        Args:
+            future (Future):
+        """
+        try:
+            response = future.result()
+
+            response_error = self._handle_response_error(response["error"])
+            if response_error:
+                # There is a server-side error sent to the client.
+                self._on_execute_error(response_error)
+                return
+
+            row_metadata = response["row_metadata"]
+            if row_metadata is not None:
+                # The result contains some rows, not an update count.
+                row_metadata = SqlRowMetadata(row_metadata)
+
+            self._on_execute_response(row_metadata, response["row_page"], response["update_count"])
+        except Exception as e:
+            # Something went bad, we couldn't get response from
+            # the server, invocation failed.
+            self._on_execute_error(self._sql_service.re_raise(e, self._connection))
+
+    def _handle_response_error(self, error):
+        """If the error is not ``None``, return it as
+        :class:`HazelcastSqlError` so that we can raise
+        it to user.
+
+        Args:
+            error (_SqlError): The error or ``None``.
+
+        Returns:
+            HazelcastSqlError: If the error is not ``None``,
+            ``None`` otherwise.
+        """
+        if error:
+            return HazelcastSqlError(error.originating_member_uuid, error.code, error.message, None)
+        return None
```

Comment:
I think we can remove the `return None` line.

Reply:
So, if I remove this, it will implicitly return `None`. I would like to be more explicit with my intent and leave this as it is